tests/integration/test_charm.py

```diff
@@ -45,7 +48,62 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
     assert len(cluster_members) == NUM_UNITS
 
     # make sure data can be written to the cluster
+    password = await get_user_password(ops_test, user=INTERNAL_USER, unit=leader_unit)
     test_key = "test_key"
     test_value = "42"
-    assert put_key(model, leader_unit, endpoints, key=test_key, value=test_value) == "OK"
-    assert get_key(model, leader_unit, endpoints, key=test_key) == test_value
+    assert (
+        put_key(
+            model,
+            leader_unit,
+            endpoints,
+            user=INTERNAL_USER,
+            password=password,
+            key=test_key,
+            value=test_value,
+        )
+        == "OK"
+    )
+    assert (
+        get_key(model, leader_unit, endpoints, user=INTERNAL_USER, password=password, key=test_key)
+        == test_value
+    )
+
+
+@pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "large"])
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_authentication(ops_test: OpsTest) -> None:
+    """Assert authentication is enabled by default.
+
+    Test reading and writing data without providing credentials.
+    Test updating the password of the internal admin user and make sure it can be used.
+    """
+    model = ops_test.model_full_name
+    endpoints = get_cluster_endpoints(ops_test, APP_NAME)
+    leader_unit = await get_juju_leader_unit_name(ops_test, APP_NAME)
+    test_key = "test_key"
+    test_value = "42"
+    new_password = "my_new_pwd"
+
+    # check that reading/writing data without credentials fails
+    assert get_key(model, leader_unit, endpoints, key=test_key) != test_value
+    assert put_key(model, leader_unit, endpoints, key=test_key, value=test_value) != "OK"
+
+    # run set-password action
```

Comment:
We should add a test case to get data with the auto generated password before we change it.

Reply:
Yes, there is, in the previous test function. We could move it here if you want to.

Reply:
True.